File: app/src/main/java/com/cleanup/todoc/domaine/usecase/CreateProjectUseCase.java
package com.cleanup.todoc.domaine.usecase;

import com.cleanup.todoc.datasource.model.ProjectEntity;
import com.cleanup.todoc.data.repository.ProjectRepository;
import com.cleanup.todoc.domaine.model.ProjectDomain;

public class CreateProjectUseCase {
    private final ProjectRepository repository;
    public CreateProjectUseCase(ProjectRepository repository){this.repository=repository;}
    public void createProject(ProjectDomain projectDomain){
        repository.insert(projectDomain);
    }
}

File: app/src/main/java/com/cleanup/todoc/domaine/usecase/CreateProjectUseCase.kt
package com.cleanup.todoc.domaine.usecase;

import com.cleanup.todoc.data.repository.ProjectRepository;
import com.cleanup.todoc.domaine.ProjectDomainUiMapper;
import com.cleanup.todoc.domaine.model.ProjectDomain;
import com.cleanup.todoc.presentation.model.Project;

public class CreateProjectUseCase {
    private final ProjectRepository repository;
    public CreateProjectUseCase(ProjectRepository repository){this.repository=repository;}
    public void createProject(Project project){
        repository.insert(ProjectDomainUiMapper.mapToDomain(project));
    }
}
